test/e2e/gotests/gotests.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 Intel Corporation.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package gotests
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/intel/pmem-csi/test/e2e/pod"
+)
+
+var _ = Describe("gotests", func() {
+	f := framework.NewDefaultFramework("gotests")
+	f.SkipNamespaceCreation = true
+
+	// Register one test for each package.
+	for _, pkg := range strings.Split(os.Getenv("TEST_PKGS"), " ") {
+		pkg := pkg
+		It(pkg, func() { runGoTest(f, pkg) })
+	}
+})
+
+// runGoTest builds and copies the Go test binary into the PMEM-CSI
+// driver container and executes it there. This way it runs in exactly
+// the same environment as the driver (the distro's kernel, our
+// container user space), which may or may not expose bugs that are
+// not found when running those tests on the build host.
+func runGoTest(f *framework.Framework, pkg string) {
+	root := os.Getenv("REPO_ROOT")
+	var err error
+
+	build := exec.Command("/bin/sh", "-c", os.Getenv("TEST_CMD")+" -c -o _work/test.test "+pkg)
```

Comment:
How about making a single binary for all unit tests and run in a single stretch.

Reply:
We would have to reorganize the test source code. I prefer to keep it decentralized (one test per package).